pkg/config/setup/config.go

```diff
@@ -2078,7 +2078,7 @@ func ResolveSecrets(config pkgconfigmodel.Config, secretResolver secrets.Compone
 func configAssignAtPath(config pkgconfigmodel.Config, settingPath []string, newValue any) error {
 	settingName := strings.Join(settingPath, ".")
 	if config.IsKnown(settingName) {
-		config.Set(settingName, newValue, pkgconfigmodel.SourceAgentRuntime)
+		config.Set(settingName, newValue, config.GetSource(settingName))
```

Comment:
Are you sure about this change ?
When using secrets it seems reasonable for the config source to be runtime, the value was not in the file itself, it was determined at runtime, and can even change during the lifetime of the agent (eg. if using secret refresh)

Reply:
It looks like we disagree on what the agent runtime layer is. The runtime layer is for values that are computed by the agent at runtime, that doesn't rely on humans, and depend on the agent's environment. `check_runners` is a good example: it depends on the number of cores the host has.

I also disagree with your take that secrets are not set by the user. Secrets are set by customers, even more so that they have to set them in their secrets manager then add the agent configuration entry to use them. The `ENC[...]` can be set in any layer like any other configuration entry. It's simply replaced at runtime, not computed. What happens if a user sets a secret entry with a different path/engine in multiple layers and need to debug?


Besides in terms of customer experience this layer doesn't really make sense. It's hard to explain and is confusing at best. Even internally it seems that we disagree on its definition. It shouldn't be surfaced to customers, thus the need to minimise what's in it.

Happy to discuss this off Github if needed